leasing/serializers/land_area.py

```diff
@@ -553,11 +553,14 @@ def create(self, validated_data):
         instance = super().create(validated_data)
         if custom_detailed_plan is not None:
             distributions = custom_detailed_plan.pop("usage_distributions", None)
-            area, created = CustomDetailedPlan.objects.update_or_create(
-                lease_area=instance, defaults={**custom_detailed_plan}
+            area = CustomDetailedPlan.objects.create(
+                lease_area_id=instance.id, **custom_detailed_plan
             )
-            if not created:
-                area.util_distributions.all().delete()
+            CustomDetailedPlan.objects.filter(
+                lease_area__in=LeaseArea.deleted_objects.filter(
+                    identifier=instance.identifier
```

Comment:
What if there are other leases that have the same lease area identifiers?

Reply:
Well I thought identifier would be somewhat unique in LeaseAreas. But if that is not the case then there is possibility that CustomDetailPlans with same LeaseArea identifiers would be deleted.